spec/calculator_spec.rb

```diff
@@ -1 +1,259 @@
-# Your code here
\ No newline at end of file
+require 'calculator'
+
+describe Calculator do
+
+
+
+	describe '#initialize' do
+
+		describe '#number_of_arguments_given' do
+    	
+	    	it 'does not raise an error without any argument' do
+	      		expect{ Calculator.new}.not_to raise_error
+	    	end
+
+	    	it 'does not raise an error passing in a bool as argument' do
+	    		stringify_param = [true, false].sample
+	      		expect{ Calculator.new(stringify_param)}.not_to raise_error
+	    	end
+
+	    end
+
+		describe 'instance_variable_checks' do
+
+	    	it 'properly set the instance params when initialized without any params' do
+	    		test_calc = Calculator.new
+	    		memory_instance = test_calc.instance_variable_get( :@memory )
+	    		stringify_instance = test_calc.instance_variable_get( :@stringify )
+	      		expect(memory_instance).to be_nil
+	      		expect(stringify_instance).to be_falsey
+	    	end
+	    	it 'properly set the instance params when initialized with a param' do
+	    		stringify_param = [true, false].sample
+	      		test_calc = Calculator.new(stringify_param)
+	    		memory_instance = test_calc.instance_variable_get( :@memory )
+	    		stringify_instance = test_calc.instance_variable_get( :@stringify )
+	      		expect(memory_instance).to be_nil
+	      		expect(stringify_instance).to eq(stringify_param)
+	    	end
+	    end
+	end
+
+	describe '#stringify' do
+
+		it 'properly set the output when stringify not provided' do
+			@test_calc = Calculator.new
+			expect(@test_calc.output(2)).not_to be_a(String)
+		end
+
+		it 'properly set the output as string when stringify set' do
+			@test_calc = Calculator.new(true)
+			expect(@test_calc.output(12)).to be_a(String)
+		end
+
+		it 'properly set the output when stringify not set' do
+			@test_calc = Calculator.new(false)
+			expect(@test_calc.output(50)).not_to be_a(String)
+		end
+	end
```

Comment:
Also, check if all the main methods(add, subtract, multiply etc) return string if `stringify` is `true`